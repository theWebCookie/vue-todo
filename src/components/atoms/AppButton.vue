<template>
  <button :class="[$style.button, customClass]" @click="handleClick">
    <slot></slot>
  </button>
</template>

<script>
export default {
  name: 'AppButton',
  props: {
    customClass: {
      type: String,
      default: '',
    },
  },
  methods: {
    handleClick(event) {
      this.$emit('click', event);
    },
  },
};
</script>

<style module lang="scss">
.button {
  all: unset;
  cursor: pointer;

  &:hover {
    color: #007bff;
  }
}
</style>
