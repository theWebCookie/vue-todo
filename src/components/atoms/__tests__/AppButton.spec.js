import { shallowMount } from '@vue/test-utils';
import { describe, it, expect } from '@jest/globals';
import AppButton from '../AppButton.vue';
import { render, fireEvent } from '@testing-library/vue';

describe('AppButton.vue', () => {
  it('renders correctly and matches the snapshot', () => {
    const wrapper = shallowMount(AppButton, {
      props: {
        customClass: 'custom-class',
      },
      slots: {
        default: 'Click Me',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('emits a click event when button is clicked', async () => {
    const { getByRole, emitted } = render(AppButton);

    const button = getByRole('button');
    await fireEvent.click(button);

    expect(emitted()).toHaveProperty('click');
    expect(emitted().click.length).toBe(2);
  });

  it('applies the active class when isActive is true', () => {
    const wrapper = shallowMount(AppButton, {
      propsData: {
        isActive: true,
      },
    });
    expect(wrapper.classes()).toContain('active');
  });

  it('does not apply the active class when isActive is false', () => {
    const wrapper = shallowMount(AppButton, {
      propsData: {
        isActive: false,
      },
    });
    expect(wrapper.classes()).not.toContain('active');
  });
});
