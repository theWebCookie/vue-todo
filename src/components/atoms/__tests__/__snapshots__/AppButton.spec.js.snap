// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton.vue renders correctly and matches the snapshot 1`] = `
<button
  class="button custom-class"
>
  
  Click Me
  
</button>
`;
